src/Microsoft.VisualStudio.Editor.Razor/DefaultTextBufferCodeDocumentProvider.cs

```diff
@@ -12,27 +12,14 @@ namespace Microsoft.VisualStudio.Editor.Razor
     [Export(typeof(TextBufferCodeDocumentProvider))]
     internal class DefaultTextBufferCodeDocumentProvider : TextBufferCodeDocumentProvider
     {
-        private readonly RazorEditorFactoryService _editorFactoryService;
-
-        [ImportingConstructor]
-        public DefaultTextBufferCodeDocumentProvider(RazorEditorFactoryService editorFactoryService)
-        {
-            if (editorFactoryService == null)
-            {
-                throw new ArgumentNullException(nameof(editorFactoryService));
-            }
-
-            _editorFactoryService = editorFactoryService;
-        }
-
         public override bool TryGetFromBuffer(ITextBuffer textBuffer, out RazorCodeDocument codeDocument)
         {
             if (textBuffer == null)
             {
                 throw new ArgumentNullException(nameof(textBuffer));
             }
 
-            if (_editorFactoryService.TryGetParser(textBuffer, out var parser) && parser.CodeDocument != null)
```

Comment:
Was a blatant mistake before to use the editor factory service here. Basically there was a chance that this call path (through the completion provider) in a valid Razor Core scenario would create all the text buffer initialization dependencies; in that case it would have tried to do that on a background thread exploding spectacularly 😄. Ultimately this code path really doesn't have any reason to want to construct the parser anyhow; it really just wants to consume it if it`s available.

Reply:
This was the fix I was going to suggest. Looks good.

Did you check if other call sites are broken by this change in semantics?

Reply:
Yup. This class today is only used (indirectly) by the directive completion provider.